src/server/pkg/log/log.go

```diff
@@ -26,16 +30,108 @@ func NewLogger(service string) Logger {
 	l.Formatter = new(prettyFormatter)
 	return &logger{
 		l.WithFields(logrus.Fields{"service": service}),
+		make(map[string]*prometheus.HistogramVec),
+		make(map[string]prometheus.Counter),
 	}
 }
 
 // Helper function used to log requests and responses from our GRPC method
 // implementations
 func (l *logger) Log(request interface{}, response interface{}, err error, duration time.Duration) {
+	state := "started"
```

Comment:
Minor thing, but it seems a little weird to compute the state in this function rather than in `ReportMetric` below, especially since you're already passing in the duration, you could pass `err` in addition and jut compute the value in there.